Component with local properties doesn't inherit service from superclass

<!--
IF YOU DON'T FILL OUT THE FOLLOWING INFORMATION WE MIGHT CLOSE YOUR ISSUE WITHOUT INVESTIGATING
-->

**I'm submitting a ...**  (check one with "x")
```
[x] bug report => search github for a similar issue or PR before submitting
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
My BaseComponent injects a service (example: https://github.com/ebrehault/component-extension-example/blob/master/src/app/base-component/base-component.component.ts#L13).
My CustomComponent inherits from BaseComponent (example: https://github.com/ebrehault/component-extension-example/blob/master/src/app/custom-component/custom-component.component.ts#L9) and it works fine, the service is properly injected.
But my CustomWithproperty component which is also inheriting from BaseComponent but defines a custom property (example: https://github.com/ebrehault/component-extension-example/blob/master/src/app/custom-with-property/custom-with-property.component.ts#L11) does not work, the service is not injected.

**Expected behavior**
The service should be injected.

**Minimal reproduction of the problem with instructions**
<!--
If the current behavior is a bug or you can illustrate your feature request better with an example, 
please provide the *STEPS TO REPRODUCE* and if possible a *MINIMAL DEMO* of the problem via
https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).
-->
Here is an example generated with Angular CLI 1.0.0:
https://github.com/ebrehault/component-extension-example

**What is the motivation / use case for changing the behavior?**
<!-- Describe the motivation or the concrete use case -->
Adding custom properties (or methods) in a component should not break superclass injected services.

Note: it looks like a regression, as it used to work with Angular 2.4

**Please tell us about your environment:**
<!-- Operating system, IDE, package manager, HTTP server, ... -->
Ubuntu 16.10.
npm 3.10.10
served with `ng serve`

* **Angular version:** 4.0.0
<!-- Check whether this is still an issue in the most recent Angular version -->

* **Browser:** [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
<!-- All browsers where this could be reproduced -->
Chrome 56
 
* **Language:** [all | TypeScript X.X | ES6/7 | ES5]
Typescript 2.2.1

* **Node (for AoT issues):** `node --version` =   
6.10.0

Yes, but we cannot use 2.0.10 with Angular 4 due to compatibility issues like`: Typescript: error node_modules/@angular/common/src/platform_id.d.ts(8,42): error TS1039: Initializers are not allowed in ambient contexts.` 

So we are blocked now! Why none recognize it as a **regression**?

--- a/packages/core/src/reflection/reflection_capabilities.ts
+++ b/packages/core/src/reflection/reflection_capabilities.ts
@@ -14,8 +14,7 @@ import {GetterFn, MethodFn, SetterFn} from './types';
 /**
  * Attention: This regex has to hold even if the code is minified!
  */
-export const DELEGATE_CTOR =
-    /^function\s+\S+\(\)\s*{\s*("use strict";)?\s*(return\s+)?(\S+\s+!==\s+null\s+&&\s+)?\S+\.apply\(this,\s*arguments\)/;
+export const DELEGATE_CTOR = /^function\s+\S+\(\)\s*{[\s\S]+\.apply\(this,\s*arguments\)/;
 
 export class ReflectionCapabilities implements PlatformReflectionCapabilities {
   private _reflect: any;

--- a/packages/core/test/reflection/reflector_spec.ts
+++ b/packages/core/test/reflection/reflector_spec.ts
@@ -8,6 +8,7 @@
 
 import {Reflector} from '@angular/core/src/reflection/reflection';
 import {DELEGATE_CTOR, ReflectionCapabilities} from '@angular/core/src/reflection/reflection_capabilities';
+import {global} from '@angular/core/src/util';
 import {makeDecorator, makeParamDecorator, makePropDecorator} from '@angular/core/src/util/decorators';
 
 interface ClassDecoratorFactory {
@@ -171,8 +172,12 @@ export function main() {
         class ChildWithCtor extends Parent {
           constructor() { super(); }
         }
+        class ChildNoCtorPrivateProps extends Parent {
+          private x = 10;
+        }
 
         expect(DELEGATE_CTOR.exec(ChildNoCtor.toString())).toBeTruthy();
+        expect(DELEGATE_CTOR.exec(ChildNoCtorPrivateProps.toString())).toBeTruthy();
         expect(DELEGATE_CTOR.exec(ChildWithCtor.toString())).toBeFalsy();
       });
     });
@@ -222,6 +227,15 @@ export function main() {
 
         class Child extends Parent {}
 
+        @ClassDecorator({value: 'child'})
+        class ChildWithDecorator extends Parent {
+        }
+
+        @ClassDecorator({value: 'child'})
+        class ChildWithDecoratorAndProps extends Parent {
+          private x = 10;
+        }
+
         // Note: We need the class decorator as well,
         // as otherwise TS won't capture the ctor arguments!
         @ClassDecorator({value: 'child'})
@@ -241,6 +255,14 @@ export function main() {
         ]);
 
         expect(reflector.parameters(Child)).toEqual([
+          [A, new ParamDecorator('a')], [B, new ParamDecorator('b')]
+        ]);
+
+        expect(reflector.parameters(ChildWithDecorator)).toEqual([
+          [A, new ParamDecorator('a')], [B, new ParamDecorator('b')]
+        ]);
+
+        expect(reflector.parameters(ChildWithDecoratorAndProps)).toEqual([
           [A, new ParamDecorator('a')], [B, new ParamDecorator('b')]
         ]);
 

